src/prefect/server/models/deployments.py

```diff
@@ -1209,3 +1217,33 @@ async def with_system_labels_for_deployment(
     ) or {}
 
     return parent_labels | default_labels | user_supplied_labels
+
+
+async def with_system_labels_for_deployment_flow_run(
```

Comment:
if its returning the labels themselves, why is this called `with_system_labels`... and not `generate` or `get_system_labels`?

Reply:
Only because that's what the corresponding method on the flow run model is called: https://github.com/PrefectHQ/prefect/blob/main/src/prefect/server/models/flow_runs.py#L615

I agree we should change the name though for both (in a separate PR).